templates/triton_services/README.md

```diff
@@ -0,0 +1,175 @@
+[Triton building complex pipeline](https://github.com/triton-inference-server/tutorials/blob/r24.04/Conceptual_Guide/Part_6-building_complex_pipelines/README.md),
+and [Triton Ray Serve Deployment](https://github.com/triton-inference-server/tutorials/tree/r24.04/Triton_Inference_Server_Python_API/examples/rayserve).
+If you need to learn more on TensorRT and Triton, please refer to their official
+documentation and tutorial.
+
+In Nvidia's tutorial, they have an image that is used to build the model and
+another image that is used to serve the model. In this tutorial, we will pull in all the
+necessary dependencies into one image. This image will be used to compile the model,
+do local development in the Anyscale Workspace, and deploy to Anyscale Services.
+
+The full `Dockerfile` used to build the image is already provided in the workspace.
+You can view the file by clicking on the `Dockerfile` in the file explorer. It shows
+what's required in the image to run this tutorial.
+
+## Compile model using Triton's Python backend on Anyscale Workspaces
+
+Unlike Nvidia's tutorial, we do not include models in the Docker image. Instead, in this
+section we will build the model and upload the model to a cloud storage such as AWS S3
+or GCP Cloud Storage for serving later.
+
+Run this code to start a Triton server using the tmp directory as the model repository.
+
+
+```python
+import tritonserver
+
+model_repository = ["/tmp/workspace/diffusion-models"]
+
+triton_server = tritonserver.Server(
+    model_repository=model_repository,
+    model_control_mode=tritonserver.ModelControlMode.EXPLICIT,
+)
+triton_server.start(wait_until_ready=True)
+```
+
+Compile the model using Triton's Python backend. This will take 10-15 minutes on a
+T4 GPU and 8-10 minutes on an A10 GPU. The model will be compiled and saved in the
+`model_repository` directory as TensorRT engine. Also keep in mind the model has to be
+built in the same type of GPU you are planning to serve the model on.
+
+
+
+```python
+import time
+import datetime
+
+
+print(f"start time: {datetime.datetime.now()}")
+t0 = time.time()
+model = triton_server.load("stable_diffusion_1_5")
+duration = time.time() - t0
+print(f"Total duration: {duration}s")
+
+# Unload the model and the server to free the memory.
+triton_server.unload(model, wait_until_unloaded=True)
+triton_server.stop()
+```
+
+After the model is compiled, we can upload the model to a cloud storage. We need to
+upload both the model config file `config.pbtxt` and the TensorRT engine model
+directory. Anyscale provides a environment variable `ANYSCALE_ARTIFACT_STORAGE` that
+can be used to store model artifacts. Use one of the following to upload the model.
+
+
+```python
+# If running in AWS
+!aws s3 cp /tmp/workspace/diffusion-models/stable_diffusion_1_5/config.pbtxt $ANYSCALE_ARTIFACT_STORAGE/triton_model_repository/stable_diffusion_1_5/config.pbtxt
```

Comment:
since this is a md, suggest removing the ! at the start of files to make copy paste easier